site/src/components/messages.js

```diff
@@ -266,6 +268,13 @@ const component = (state, emit) => {
     }
   }
 
+  if (state.messages.isScrolledToBottom()) {
+    // scroll to bottom after re-render
+    setTimeout(() => {
+      state.messages.el.scrollTop = state.messages.el.scrollHeight + 999
+    }, 50)
+  }
+
```

Comment:
Why the `+ 999`? Shouldn't `scrollHeight` automatically adjust enough?

Reply:
Yep, but it didn't.

JS is bad.